internal/controllers/machineset/machineset_controller_status_test.go

```diff
@@ -0,0 +1,625 @@
+				createMachine("machine-1", readyCondition),
+				createMachine("machine-2", readyCondition),
+			},
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.MachineSetMachinesReadyV1Beta2Condition,
+				Status: metav1.ConditionTrue,
+				Reason: v1beta2conditions.MultipleInfoReportedReason,
+			},
+		},
+		{
+			name:       "one ready, one has nothing reported",
+			machineSet: machineSet,
+			machines: []*clusterv1.Machine{
+				createMachine("machine-1", readyCondition),
+				createMachine("machine-2"),
+			},
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetMachinesReadyV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  v1beta2conditions.NotYetReportedReason,
+				Message: "Condition Ready not yet reported from Machine machine-2",
+			},
+		},
+		{
+			name:       "one ready, one reporting not ready, one reporting unknown, one reporting deleting",
+			machineSet: machineSet,
+			machines: []*clusterv1.Machine{
+				createMachine("machine-1", readyCondition),
+				createMachine("machine-2", metav1.Condition{
+					Type:    clusterv1.MachineReadyV1Beta2Condition,
+					Status:  metav1.ConditionFalse,
+					Reason:  "SomeReason",
+					Message: "HealthCheckSucceeded: Some message",
+				}),
+				createMachine("machine-3", metav1.Condition{
+					Type:    clusterv1.MachineReadyV1Beta2Condition,
+					Status:  metav1.ConditionUnknown,
+					Reason:  "SomeUnknownReason",
+					Message: "Some unknown message",
+				}),
+				createMachine("machine-4", metav1.Condition{
+					Type:    clusterv1.MachineReadyV1Beta2Condition,
+					Status:  metav1.ConditionFalse,
+					Reason:  clusterv1.MachineDeletingV1Beta2Reason,
+					Message: "Deleting: Machine deletion in progress, stage: DrainingNode",
+				}),
+			},
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetMachinesReadyV1Beta2Condition,
+				Status:  metav1.ConditionFalse,
+				Reason:  v1beta2conditions.MultipleIssuesReportedReason,
+				Message: "Deleting: Machine deletion in progress, stage: DrainingNode from Machine machine-4; HealthCheckSucceeded: Some message from Machine machine-2; Some unknown message from Machine machine-3",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			setMachinesReadyCondition(ctx, tt.machineSet, tt.machines, tt.machines != nil)
+
+			condition := v1beta2conditions.Get(tt.machineSet, clusterv1.MachineSetMachinesReadyV1Beta2Condition)
+			g.Expect(condition).ToNot(BeNil())
+			g.Expect(*condition).To(v1beta2conditions.MatchCondition(tt.expectCondition, v1beta2conditions.IgnoreLastTransitionTime(true)))
+		})
+	}
+}
+
+func Test_setMachinesUpToDateCondition(t *testing.T) {
+	machineSet := &clusterv1.MachineSet{}
+
+	tests := []struct {
+		name            string
+		machineSet      *clusterv1.MachineSet
+		machines        []*clusterv1.Machine
+		expectCondition metav1.Condition
+	}{
+		{
+			name:       "getAndAdoptMachines failed",
```

Comment:
We probably should also have a case with no Machines

Let's also add a case where the unknown condition is not set on the machine (because that is our current state + it can happen until the controllers set the up-to-date condition)

Would be also good to have a case with multiple Machines